src/FixedRateIrm.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import {IIrm} from "../lib/morpho-blue/src/interfaces/IIrm.sol";
+import {IFixedRateIrm} from "./interfaces/IFixedRateIrm.sol";
+
+import {MarketParamsLib} from "../lib/morpho-blue/src/libraries/MarketParamsLib.sol";
+import {Id, MarketParams, Market} from "../lib/morpho-blue/src/interfaces/IMorpho.sol";
+
+/// @title FixedRateIrm
+/// @author Morpho Labs
+/// @custom:contact security@morpho.org
+contract FixedRateIrm is IFixedRateIrm {
+    using MarketParamsLib for MarketParams;
+
+    /* EVENTS */
+
+    /// @notice Emitted when a borrow rate is set.
+    event SetBorrowRate(Id indexed id, uint256 newBorrowRate);
+
+    /* ERRORS */
+
+    /// @dev Thrown when the rate is not already set for this market.
+    string public constant RATE_NOT_SET = "rate not set";
+    /// @dev Thrown when the rate is already set for this market.
+    string public constant RATE_SET = "rate set";
+    /// @dev Thrown when trying to set the rate at zero.
+    string public constant RATE_ZERO = "rate zero";
+
+    /* STORAGE */
+
+    /// @notice Borrow rates.
+    mapping(Id => uint256) public _borrowRate;
```

Comment:
```suggestion
    mapping(Id => uint256) internal _borrowRate;
```

Is it useful?

Reply:
yes it is (imagine you want to know if the rate has already been set without a try catch)